packages/core/src/components/essentials/threshold.ts

```diff
@@ -0,0 +1,144 @@
+import { ChartModel } from "../../model";
+import { AxisPositions, Events } from "../../interfaces";
+import { select, mouse } from "d3-selection";
+
+// Carbon position service
+import Position, { PLACEMENTS } from "@carbon/utils-position";
+
+// import the settings for the css prefix
+import settings from "carbon-components/src/globals/js/settings";
+
+export class Threshold extends Component {
+	type = "threshold";
+
+	threshold: any;
+	thresholdClass: string;
+	thresholdIdentifierClass: string;
+
+	label: any;
+
+	positionService = new Position();
+
+	constructor(model: ChartModel, services: any, configs: any) {
+		super(model, services, configs);
+	}
+
+	render() {
+		const { value, fillColor, axisPosition, index } = this.configs;
+		const chartprefix = Tools.getProperty(this.model.getOptions(), "style", "prefix");
+		this.thresholdClass = `${settings.prefix}--${chartprefix}--threshold`;
+		this.thresholdIdentifierClass = `${axisPosition}-${index}`;
+
+		this.threshold = DOMUtils.appendOrSelect(this.parent, `g.${this.thresholdClass}.${this.thresholdIdentifierClass}`).raise();
+		const thresholdRect = DOMUtils.appendOrSelect(this.threshold, `rect.threshold-hoverable-area`);
+		const thresholdLine = DOMUtils.appendOrSelect(this.threshold, `line`);
+
+		thresholdLine.style("stroke", fillColor);
+
+		const scale = this.services.cartesianScales.getScaleByPosition(axisPosition);
+		const isVertical = [AxisPositions.LEFT, AxisPositions.RIGHT].includes(axisPosition);
+		const mainXScale = this.services.cartesianScales.getMainXScale();
+		const mainYScale = this.services.cartesianScales.getMainYScale();
+		const [xScaleStart, xScaleEnd] = mainXScale.range();
+		const [yScaleEnd, yScaleStart] = mainYScale.range();
+
+		if (isVertical) {
+			const y = scale(value) - yScaleStart;
+			this.threshold.attr("transform", `translate(${xScaleStart}, ${y})`);
+			thresholdLine.attr("x2", xScaleEnd - xScaleStart);
+			thresholdRect.attr("width", xScaleEnd - xScaleStart);
+		} else {
+			const x = scale(value);
+			this.threshold.attr("transform", `translate(${x}, 0)`);
+			thresholdLine.attr("y2", yScaleEnd);
+			thresholdRect.attr("width", yScaleEnd - yScaleStart);
+			thresholdRect.classed("rotate", true);
+		}
+
+		const self = this;
+		this.services.events.addEventListener(Events.Threshold.SHOW, e => {
+			const hovered = e.detail.hoveredElement.node();
+			if (hovered === self.threshold) {
+				this.setThresholdLabelPosition();
+				this.label.classed("hidden", false);
+			}
+		});
+
+		this.services.events.addEventListener(Events.Threshold.HIDE, e => {
+			this.label.classed("hidden", true);
+		});
+
+		this.appendThresholdLabel();
+
+		this.addEventListeners();
+	}
+
+	appendThresholdLabel() {
+		const holder = select(this.services.domUtils.getHolder());
+		const { value, valueFormatter, fillColor, label = "Threshold" } = this.configs;
+		const formattedValue = valueFormatter ? valueFormatter(value) : value.toLocaleString("en");
```

Comment:
Is there any way to take the default formatter we actually use in the corresponding axis tick labels?